### PR TITLE
Issue 1330

### PR DESCRIPTION
**Pull request by Naupio**
## What problem does this PR solve?
issue: #1330 
- add sourcify metadata for smart contract
- fix sourcify process worker
- add new field sourcify metadata of  smart contract object 
- add testing code
- add history process task
## Check List
- smart contract with sourcify metadate
- history data was process
- auto worker process correct when smart contract register
#### Test
- unit test
#### Task
 - none

### CODE REVIEW

File: lib/godwoken_explorer/graphql/sourcify.ex [hunk 1]
```diff
@@ -455,3 +452,12 @@ defmodule GodwokenExplorer.Graphql.Sourcify do
 
   def failed_verification_message, do: @failed_verification_message
 end
+
+defmodule GodwokenExplorer.Graphql.SourcifyExport do
```
Comment (zmcNotafraid) [inline lib/godwoken_explorer/graphql/sourcify.ex:456]: 这里起个新文件不更好吗

File: lib/godwoken_explorer/smart_contract.ex [hunk 2]
```diff
@@ -67,6 +67,8 @@ defmodule GodwokenExplorer.SmartContract do
     field :transaction_count, :integer, virtual: true
     field :eth_address, :binary, virtual: true
 
+    field :sourcify_metadata, {:array, :map}
```
Comment (zmcNotafraid) [inline lib/godwoken_explorer/smart_contract.ex:70]: 你这里明显是再创建一张表，然后1对多的关系绑定。
参考下 blockscout 怎么设计的

**Review by zmcNotafraid — CHANGES_REQUESTED** [review]

File: lib/godwoken_explorer/graphql/sourcify.ex (hunk 1 above)
Comment (Naupio) [reply lib/godwoken_explorer/graphql/sourcify.ex]: fixed

File: lib/godwoken_explorer/smart_contract.ex (hunk 2 above)
Comment (Naupio) [reply lib/godwoken_explorer/smart_contract.ex:70]: 如讨论, 这里维持这种方式

**Review by zmcNotafraid — APPROVED** [review]

**Review by FrederLu — APPROVED** [review]

**Pull request merged**